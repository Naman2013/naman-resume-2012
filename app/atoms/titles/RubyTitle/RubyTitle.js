import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './RubyTitle.style';

const RubyTitle = ({ text }) => (
  <Fragment>
    <h3 className="title">{text}</h3>
    <style jsx>{style}</style>
  </Fragment>
);

RubyTitle.propTypes = {
  text: PropTypes.string.isRequired,
};

export default RubyTitle;
