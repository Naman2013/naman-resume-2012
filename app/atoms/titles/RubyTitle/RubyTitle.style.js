import css from 'styled-jsx/css';
import { resetMarginPadding } from 'app/styles/variables/utils';
import { primaryFont } from 'app/styles/variables/fonts';
import { astronaut } from 'app/styles/variables/colors_tiles_v4';

export default css`
  .title {
    ${resetMarginPadding}
    font-family: ${primaryFont};
    color: ${astronaut};
    font-weight: 800;
    letter-spacing: 1.5px;
    text-transform: uppercase;
    font-size: 12px;
  }
`;
