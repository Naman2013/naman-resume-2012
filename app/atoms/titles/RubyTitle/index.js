import RubyTitle from './RubyTitle';

export default RubyTitle;
