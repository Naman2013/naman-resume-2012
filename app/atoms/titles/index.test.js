import React from 'react';
import { shallow } from 'enzyme';
import RubyTitle from './RubyTitle';

describe('RubyTitle', () => {
  const shallowWrapper = shallow(<RubyTitle text="Test" />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
