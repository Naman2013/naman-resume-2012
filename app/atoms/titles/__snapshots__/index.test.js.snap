// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubyTitle should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RubyTitle
    text="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3
          className="jsx-3247648496 title"
        >
          Test
        </h3>,
        <JSXStyle
          css={
            Array [
              ".title.jsx-3247648496{
  margin: 0;
  padding: 0;
 font-family:brandon-grotesque, sans-serif;color:#415671;font-weight:800;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;font-size:12px;}",
            ]
          }
          styleId="3247648496"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
          "className": "jsx-3247648496 title",
        },
        "ref": null,
        "rendered": "Test",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".title.jsx-3247648496{
  margin: 0;
  padding: 0;
 font-family:brandon-grotesque, sans-serif;color:#415671;font-weight:800;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;font-size:12px;}",
          ],
          "styleId": "3247648496",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3
            className="jsx-3247648496 title"
          >
            Test
          </h3>,
          <JSXStyle
            css={
              Array [
                ".title.jsx-3247648496{
  margin: 0;
  padding: 0;
 font-family:brandon-grotesque, sans-serif;color:#415671;font-weight:800;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;font-size:12px;}",
              ]
            }
            styleId="3247648496"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
            "className": "jsx-3247648496 title",
          },
          "ref": null,
          "rendered": "Test",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".title.jsx-3247648496{
  margin: 0;
  padding: 0;
 font-family:brandon-grotesque, sans-serif;color:#415671;font-weight:800;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;text-transform:uppercase;font-size:12px;}",
            ],
            "styleId": "3247648496",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
